src/averageAges.js

```diff
@@ -14,12 +14,18 @@
  *
  * @return {number}
  */
+function calcAverageAge(people) {
+  return people
+    .map(person => person.died - person.born)
+    .reduce((a, b) => a + b, 0) / people.length;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredMen = people.filter(person => (!century
+    || century === Math.ceil(person.died / 100))
+    && person.sex === 'm');
+
+  return calcAverageAge(filteredMen);
```

Comment:
```suggestion
const { filteredMenCount, menSumAge } = people.reduce((acc, person) => {
    const { sex, born, died } = person;
    const isPerson = century
      ? sex === 'm' && Math.ceil(died / 100) === century
      : sex === 'm';

    return isPerson ? {
      menSumAge: acc.menSumAge + died - born, filteredMenCount: acc.menCount + 1,
    } : acc;
  }, {
    menSumAge: 0,
    filteredMenCount: 0,
  });
```

Reply:
Goof job.
I just want to show how to improve performance and instead two iteraations use one iteration with powerful method reduce.